app/gui/view/graph-editor/src/component/node/input/widget.rs

```diff
@@ -0,0 +1,163 @@
+//! Definition all variants of possible node widgets and common widget API.
+
+use crate::prelude::*;
+
+use enso_frp as frp;
+use ensogl::application::Application;
+use ensogl::data::color;
+use ensogl::display;
+use ensogl::display::object::event;
+use ensogl_component::drop_down::Dropdown;
+
+
+
+// ==================
+// === NodeWidget ===
+// ==================
+
+ensogl::define_endpoints_2! {
+    Input {
+        set_current_value(Option<ImString>),
+        set_focused(bool),
+    }
+    Output {
+        value_changed(Option<ImString>),
+    }
+}
+
+/// Possible widgets for a node input.
+#[derive(Clone, Debug, CloneRef)]
+pub enum NodeWidget {
+    /// A widget for selecting a single value from a list of available options.
+    SingleChoice(SingleChoice),
+}
+
+impl NodeWidget {
+    /// Create a new node widget, selecting the appropriate widget type based on the provided
+    /// argument info.
+    pub fn new(
+        app: &Application,
+        argument_info: span_tree::ArgumentInfo,
+        node_height: f32,
+    ) -> Option<Self> {
+        // TODO: support more widgets, use engine provided widget type
```

Comment:
1. Missing dot at the end.
2. TODO left in the code without link to issue and info who left it.